gateway/gateway_test.go

```diff
@@ -352,9 +352,11 @@ func TestGatewayGet(t *testing.T) {
 	}{
 		{"127.0.0.1:8080", "/", http.StatusNotFound, "404 page not found\n"},
 		{"127.0.0.1:8080", "/" + k.Cid().String(), http.StatusNotFound, "404 page not found\n"},
+		{"127.0.0.1:8080", "/ipfs/this-is-not-a-cid", http.StatusInternalServerError, "failed to resolve /ipfs/this-is-not-a-cid: invalid path \"/ipfs/this-is-not-a-cid\": invalid CID: illegal base32 data at input byte 3\n"},
```

Comment:
I really don't like this tests where we compare the body for errors. Error information may change slightly depending on the backing implementation. I think rewriting these tests can be part of #156 or even #183.

Reply:
@hacdias these don't do much for UX, but also we dont expect these messages to change often.

Once set in stone, make good canaries to hint someone in transitive dependencies messed up refactor :) 
I would keep this, at least until go-libipfs/gateway and rhea work is in progress – better to have more tests than less.

I remember in the past an external library (like go-cid or go-multihash/base, don't remember which one) changed something, and we detected it in Kubo, 2 layers later, because a specific error message changed to a generic one.

ps. At some point we will add more friendly HTML error responses if a request was made with `Accept` that includes `text/html` (web browser), like we did for dag-cbor.